### DISCUSSION TITLE
[Feature Request] ElContextMenu (右键菜单组件 )

### DISCUSSION THREAD

**Discussion by xiaoxian521** [element-plus/element-plus#909] [Ideas]
we want to add a right-click menu component as https://heynext.github.io/v-contextmenu/

**Comment by jw-foss** [comment]
So as I mentioned in #910 , we currently have no human resources to do these heavy, non atomic components.

Also I wanted to extract these components out of the scope of `ElementPlus`, `ElementPlus` is supposed to be an atomic component library, for these complex components I think we can publish them under `@element-plus` scope. I have created a repo for this.

Are you interested in participant this incredible idea? I mean you can be the one who builds this component upon `ElementPlus`

The repo you can find it here. https://github.com/element-plus/el-context-menu

If you are willing to do this, just let me know.

**Reply by xiaoxian521** [reply]
Of course, but I need to know what specifications this component needs to integrate into element-plus better

**Reply by jw-foss** [reply]
You might get the exact same API from the library you mentioned above.
As for the UI part, I will ask our designer to take a look on this. Before that, you might just improvise as you wanted.

**Reply by jw-foss** [reply]
I want you to pull request in the repo I mentioned. You can get started with these starter kit that we provided. Ping me if you need any assistance.

**Reply by xiaoxian521** [reply]
In that case, I might have a development style of my own, but I would label it as el-menu

**Reply by jw-foss** [reply]
You can build the component upon `el-menu` that will save you some time. The control is all on you also the credits, it's just like some sub repo under this org.

**Reply by jw-foss** [reply]
Let's get started!

**Reply by xiaoxian521** [reply]
I change how to contribute code to  https://github.com/element-plus/el-context-menu, through PR or ?

**Comment by carlosyan1807** [comment]
any news?

**Comment by eflogic** [comment]
各位大佬，翘首以盼ContextMenu组件  :D

**Reply by eflogic** [reply]
参考
https://github.com/fkhadra/react-contexify

**Comment by Abhay2133** [comment]
Hi @jw-foss I like to work on this context menu feature, is there any internal docs available , sorry didn't found found any on readme of project ?

**Reply by Dsaquel** [reply]
You can try virtual-triggering from el-dropdown [docs](https://element-plus.org/en-US/component/dropdown#virtual-triggering).
